### PR TITLE
feat(gaussdb): add proxy resource support

### LINKED ISSUE #1577: [Gaussdb] Add seperated proxy resource for mysql instance
I'd like to add seperated proxy resource for mysql instance

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1577 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run=TestAccGaussDBProxy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run=TestAccGaussDBProxy_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBProxy_basic
=== PAUSE TestAccGaussDBProxy_basic
=== CONT  TestAccGaussDBProxy_basic
--- PASS: TestAccGaussDBProxy_basic (1530.40s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1530.430s
```

### CODE REVIEW

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	client, err := config.GaussdbV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud GaussDB client: %s ", err)
+	}
+
+	if d.HasChange("node_num") {
+		oldnum, newnum := d.GetChange("node_num")
+		if newnum.(int) < oldnum.(int) {
+			return fmtp.DiagErrorf("Error updating gaussdb_mysql_proxy %s: new node num should be greater than old num", d.Id())
+		}
```
Comment (Lance52259) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go:121]: If the `node_num` does not support shrink, maybe we can bring this logic to the schema and provide a `DiffSuppressFunc` behavior to ignore the update operation?

Comment (niuzhenguo) [reply huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go:121]: should have a validation function for this.

Comment (niuzhenguo) [reply huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go:121]: validation doesn't support `old` value...

**Pull request merged**